python/ray/dag/compiled_dag_node.py

```diff
@@ -336,6 +351,25 @@ def __init__(
                     input_channel_to_idx[channel] = channel_idx
 
                 task_input = _ExecutableTaskInput(arg, channel_idx)
+            elif isinstance(arg, list):
+                # Handle the case where the argument is a list of channels.
+                channel_idx_list = []
+                for item in arg:
+                    if isinstance(item, ChannelInterface):
+                        channel = item
+
+                        if channel in input_channel_to_idx:
```

Comment:
This is the same as lines 344 to 351. I'm considering adding a function for it, but the function would update both its input arguments (`input_channel_to_idx`) and class fields such as `self.input_channels`. This doesn’t seem like appropriate behavior for a helper function. I would appreciate more input on this.